src/agent/agent.c

```diff
@@ -164,6 +164,10 @@ static int agent_setup_heartbeat_timer(Agent *agent) {
 
         assert(agent);
 
+        if (agent->heartbeat_interval_msec <= 0) {
+                return 0;
+        }
+
```

Comment:
Hmm, wouldn't it be nice to have a warning in the logs that heartbeat is disabled? Maybe as a part of configuration phase?

Reply:
Done :heavy_check_mark: